sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/meta/provider/kafka/KafkaTableProvider.java

```diff
@@ -69,65 +58,16 @@ public BeamSqlTable buildBeamSqlTable(Table table) {
       topics.add(topic.toString());
     }
 
-    PayloadFormat payloadFormat =
-        properties.containsKey("format")
-            ? PayloadFormat.valueOf(properties.getString("format").toUpperCase())
-            : PayloadFormat.CSV;
-
-    switch (payloadFormat) {
-      case CSV:
-        return new BeamKafkaCSVTable(schema, bootstrapServers, topics);
-      case AVRO:
-        return new BeamKafkaAvroTable(schema, bootstrapServers, topics);
-      case JSON:
-        return new BeamKafkaJsonTable(schema, bootstrapServers, topics);
-      case PROTO:
-        return protoTable(schema, bootstrapServers, topics, properties);
-      case THRIFT:
-        return thriftTable(schema, bootstrapServers, topics, properties);
-      default:
-        throw new IllegalArgumentException("Unsupported payload format: " + payloadFormat);
-    }
-  }
-
-  private BeamKafkaProtoTable protoTable(
-      Schema schema, String bootstrapServers, List<String> topics, JSONObject properties) {
-    String protoClassName = properties.getString("protoClass");
-    try {
-      Class<?> protoClass = Class.forName(protoClassName);
-      return new BeamKafkaProtoTable(schema, bootstrapServers, topics, protoClass);
-    } catch (ClassNotFoundException e) {
-      throw new IllegalArgumentException("Incorrect proto class provided: " + protoClassName);
-    }
-  }
-
-  private BeamKafkaThriftTable thriftTable(
-      Schema schema, String bootstrapServers, List<String> topics, JSONObject properties) {
-    final String thriftClassName = properties.getString("thriftClass");
-    final String thriftProtocolFactoryClassName =
-        properties.getString("thriftProtocolFactoryClass");
-    try {
-      final Class<TBase> thriftClass = (Class<TBase>) Class.forName(thriftClassName);
-      final TProtocolFactory thriftProtocolFactory;
-      try {
-        final Class<TProtocolFactory> thriftProtocolFactoryClass =
-            (Class<TProtocolFactory>) Class.forName(thriftProtocolFactoryClassName);
-        thriftProtocolFactory = thriftProtocolFactoryClass.getDeclaredConstructor().newInstance();
-      } catch (ClassNotFoundException e) {
-        throw new IllegalArgumentException(
-            "Incorrect thrift protocol factory class provided: " + thriftProtocolFactoryClassName);
-      } catch (InstantiationException
-          | IllegalAccessException
-          | InvocationTargetException
-          | NoSuchMethodException e) {
-        throw new IllegalStateException(
-            "Could not instantiate the thrift protocol factory class", e);
-      }
-      return new BeamKafkaThriftTable<>(
-          schema, bootstrapServers, topics, thriftClass, thriftProtocolFactory);
-    } catch (ClassNotFoundException e) {
-      throw new IllegalArgumentException("Incorrect thrift class provided: " + thriftClassName);
+    String payloadFormat =
+        properties.containsKey("format") ? properties.getString("format") : "csv";
+    if (payloadFormat.equals("csv")) {
```

Comment:
Can you add a comment here to describe why CSV is a special why that does not go through PayloadSerializerKafkaTable?

Reply:
Done.